heron/spi/src/java/com/twitter/heron/spi/packing/PackingPlan.java

```diff
@@ -253,6 +279,13 @@ public int hashCode() {
       return result;
     }
 
+    /**
+     * Returns a {@link ContainerPlan} with updated scheduledResource
+     */
+    public ContainerPlan cloneWithScheduledResource(Resource resource) {
```

Comment:
Just because nobody else currently needs it besides the class itself. I typically keep things as restrictive as necessary and increase visibility when necessary, or when ready to provide a public, stable API.